docs/code-style.md

```diff
@@ -0,0 +1,45 @@
+# KubeAdmiral Code Style
+
+## Linting
+
+KubeAdmiral uses [golangci-lint](https://golangci-lint.run/usage/install/) for linting. All code should adhere to the rules specified in the [configuration file](../.golangci.yml).
+
+In order to check if your code adheres to our rules, you may use golangci-lint. First, ensure that you have golangci-lint installed. Then, run the following command:
+
+```console
+$ golangci-lint run --new-from-rev=origin/develop --timeout=2m --max-issues-per-linter=10 --go=1.19 --whole-files
+```
+
+## Logging Conventions
+
+Logs play an important role when troubleshooting issues. Thus, it is also important to ensure that our logs are readable, navigable and doesn't devolve into noise.
+
+In order to achieve this, KubeAdmiral follows a set of logging conventions described below.
+
+1. **ALWAYS** use [klog](https://pkg.go.dev/k8s.io/klog/v2) for logging. Klog is the de facto logging package for Kubernetes development.
+
+2. Use contextual logging whenever possible. **ALWAYS** use `klog.FromContext` to obtain a logger whenever there is a `Context` available.
+
+3. Consequently, top-level controllers or reconciliation workers **MUST** inject a logger into its `Context` using `klog.NewContext` before passing it to any nested calls. The injected logger **MUST** have the following values:
+
+    1. `controller`: indicating the name of the controller
+    2. `object`: indicating the qualified name of the object being reconciled
+
+4. Log messages **MUST** have the following format:
+
+    1. **ALWAYS** begin with a capital letter (with the exception of identifiers that conventionally begin with a lowercase)
```

Comment:
nit: "ALWAYS" is not an RFC2119 keyword.

Reply:
replaced use of "ALWAYS" with RFC2119 keywords